src/AutoRest.CSharp/LowLevel/Output/ConvenienceMethod.cs

```diff
@@ -2,11 +2,12 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System.Collections.Generic;
+using AutoRest.CSharp.Common.Input;
 using AutoRest.CSharp.Generation.Types;
 using AutoRest.CSharp.Output.Models.Requests;
 using AutoRest.CSharp.Output.Models.Shared;
 
 namespace AutoRest.CSharp.Output.Models
 {
-    internal record ConvenienceMethod(MethodSignature Signature, IReadOnlyList<(Parameter Protocol, Parameter? Convenience)> ProtocolToConvenienceParameters, CSharpType? ResponseType, Diagnostic? Diagnostic);
+    internal record ConvenienceMethod(MethodSignature Signature, IReadOnlyList<(Parameter Protocol, Parameter? Convenience, InputParameter? Input)> ProtocolToConvenienceParameters, CSharpType? ResponseType, Diagnostic? Diagnostic);
```

Comment:
`ConvenienceMethod` should be decoupled with the information from code model.

Reply:
`Parameter` is just a light present of a model, it just retrieve out the type name, no detail schema information.  In order to support flatten alias to parameters, and combine the parameters as a request content, we need the detail schema information. If we add those into Parameter, then Paramter will be almost the same as `InputParameter`.
Yes, I agree it is better to decouple with input type in the ConvenienceMethod, but now, we will remain the current design of `Parameter` and may think detail about how to design Parameter and decouple this later.

Reply:
I don't mean to change `Parameter`. I'm saying we decouple `InputParameter` from `ConvenienceMethod`. Seems only type information of `InputParameter` is used? So we could only store the type information to `ConvenienceMethod`?

Reply:
Except the `InputType`, the `Kind` is also needed. And if we add `InputType` and `Kind` into the parameter chain, it is not better to add `InputParameter` directory. Both `InputType` and `Kind` are the in InputModel serial, also